ltdmason/manifest.py

```diff
@@ -0,0 +1,108 @@
+"""Interface for stack manifest YAML files written by buildlsstsw.sh
+
+The purpose of these YAML files is to isolate ltd-mason from Eups/Scons
+builds of the software itself and to merely tell ltd-mason where the built
+software can be found, and metadata about the versioning of this Stack.
+"""
+
+from __future__ import (division, absolute_import, print_function,
+                        unicode_literals)
+from builtins import *  # NOQA
+from future.standard_library import install_aliases
+install_aliases()
+
+from urllib.parse import urlparse
+import os
+
+import ruamel.yaml
+
+
+class Manifest(object):
+    """Representation of a YAML-encoded manifest for an LSST stack product.
+
+    Parameters
+    ----------
+    yaml_data : str
+        Stack manifest, encoded as YAML text.
+
+    Attributes
+    ----------
+    data : `ruamel.yaml` object
+        Manifest dataset loaded by :mod:`ruamel.yamel`.
+    """
+    def __init__(self, f):
+        super().__init__()
+        self.data = ruamel.yaml.load(f, ruamel.yaml.RoundTripLoader)
```

Comment:
I think I can actually co-opt http://json-schema.org to also validate YAML structures using https://github.com/Julian/jsonschema